--- build-aux/ltmain.sh.unpatched
+++ build-aux/ltmain.sh
@@ -1220,6 +1220,60 @@
 
 
 
+## ------------------ ##
+##    libi8x hack.    ##
+## ------------------ ##
+
+# func_fatal_libi8x arg...
+# Echo program name prefixed message to standard error, followed by
+# a request for a bug report, and exit.
+func_fatal_libi8x ()
+{
+    func_error ${1+"$@"}
+    func_error "Please report this to infinity@sourceware.org"
+    func_fatal_error "Fatal configuration error."
+}
+
+case $version_type in
+  *linux)
+    # Transform the coded name of the library:
+    # - libi8x-0.0.5.so.1
+    # + libi8x.so.1
+    case "X$soname_spec" in
+      "X\$libname\$release\$shared_ext\$major")
+        soname_spec="\$libname\$shared_ext\$major"
+        ;;
+      "X\${libname}\${release}\${shared_ext}\$major")
+        soname_spec="\${libname}\${shared_ext}\$major"
+        ;;
+      *)
+        func_fatal_libi8x "Unexpected soname_spec \`$soname_spec'"
+        ;;
+    esac
+
+    # Transform the list of archive names:
+    # - libi8x-0.0.5.so.1.0.0   libi8x-0.0.5.so.1   libi8x.so
+    # + libi8x-0.0.5.so         libi8x.so.1         libi8x.so
+    case "X$library_names_spec" in
+      "X\$libname\$release\$shared_ext\$versuffix \$libname\$release\$shared_ext\$major \$libname\$shared_ext")
+        library_names_spec="\$libname\$release\$shared_ext $soname_spec \$libname\$shared_ext"
+	;;
+      "X\${libname}\${release}\${shared_ext}\$versuffix \${libname}\${release}\${shared_ext}\$major \$libname\${shared_ext}")
+        library_names_spec="\${libname}\${release}\${shared_ext} $soname_spec \$libname\${shared_ext}"
+	;;
+      *)
+	func_fatal_libi8x "Unexpected library_names_spec \`$library_names_spec'"
+	;;
+    esac
+    ;;
+  *)
+    func_fatal_libi8x "Unknown library version type \`$version_type'"
+    ;;
+esac
+
+
+
+
 ## ----------- ##
 ##    Main.    ##
 ## ----------- ##
@@ -8566,10 +8620,18 @@
 	fi
 
 	# Create links to the real library.
+	# libi8x hack: Unpatched libtool makes these links:
+	#   libi8x.so   -> libi8x-0.0.6.so
+	#   libi8x.so.1 -> libi8x-0.0.6.so
+	# We want these links:
+	#   libi8x.so   -> libi8x.so.1
+	#   libi8x.so.1 -> libi8x-0.0.6.so
+	i8x_target="$realname"
 	for linkname in $linknames; do
-	  if test "$realname" != "$linkname"; then
-	    func_show_eval '(cd "$output_objdir" && $RM "$linkname" && $LN_S "$realname" "$linkname")' 'exit $?'
+	  if test "$i8x_target" != "$linkname"; then
+	    func_show_eval '(cd "$output_objdir" && $RM "$linkname" && $LN_S "$i8x_target" "$linkname")' 'exit $?'
 	  fi
+	  i8x_target="$linkname"
 	done
 
 	# If -module or -export-dynamic was specified, set the dlname.
